platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventIntake.java

```diff
@@ -190,6 +238,14 @@ private static boolean isNotConsensus(final EventImpl event) {
      */
     private void handleConsensus(final ConsensusRound consensusRound) {
         if (consensusRound != null) {
+
+            // If we are asynchronously prehandling transactions, we need to
+            // wait for prehandles to finish before proceeding. It is critically
+            // important that prehandle is always called prior to handleConsensusRound().
+            if (prehandlePool != null) {
+                consensusRound.forEach(e -> ((EventImpl) e).awaitPrehandleCompletion());
```

Comment:
The amount of time waited for this should be measured

Reply:
added metric